tests/jballerina-unit-test/src/test/resources/test-src/types/future/future_positive.bal

```diff
@@ -0,0 +1,180 @@
+function testBasicTypes() {
+    future<int> f1 = start add(5, 2);
+    future<boolean> f2 = start status();
+    future<string> f3 = start concat("foo");
+
+    int result1 = wait f1;
+    boolean result2 = wait f2;
+    string result3 = wait f3;
+
+    assertEquality(7, result1);
+}
+
+function testBasicTypesWithoutFutureConstraint() {
+    future f1 = start add(5, 2);
+    future f2 = start status();
+    future f3 = start concat("foo");
+
+    any|error result1 = wait f1;
+    any|error result2 = wait f2;
+    any|error result3 = wait f3;
+
+    assertEquality(7, result1); 
+}
+
+function testRefTypes() {
+    future<xml> a = start xmlFile();
+    future<json> b = start jsonObject();
+
+    xml x = wait a;
+    json y = wait b;
+    
+    assertEquality("5", y.toString());
+}
+
+function testRefTypesWithoutFutureConstraint() {
+    future a = start xmlFile();
+    future b = start jsonObject();
+
+    any|error x = wait a;
+    any|error y = wait b;
+    
+    assertEquality("5", y.toString());
```

Comment:
IMO wherever possible it's always better to test against the exact value instead of a `toString()` representation since we'll also validate the type.

```suggestion
    assertEquality(5, y);
```

Reply:
Fixed